autoload/rust.vim

```diff
@@ -208,6 +280,23 @@ function! rust#AppendCmdLine(text)
 	return cmd
 endfunction
 
+" Tokenize the String according to shell parsing rules
+function! s:ShellTokenize(text)
```

Comment:
I’m not confident about this function across different shells. The most likely culprit for problems is Windows, which has a very basic shell as far as things like interpolation and quotation marks are concerned—has it been tested there?